agent/src/containerconfigparser.h

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2016-2017 Pelagicore AB
+ *
+ * Permission to use, copy, modify, and/or distribute this software for
+ * any purpose with or without fee is hereby granted, provided that the
+ * above copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
+ * WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR
+ * BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES
+ * OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS,
+ * WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,
+ * ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS
+ * SOFTWARE.
+ *
+ * For further information see LICENSE
+ */
+
+#pragma once
+
+#include "softwarecontainer-common.h"
+#include "softwarecontainererror.h"
+
+namespace softwarecontainer {
+
+class ContainerConfigParseError
+{
+public:
+    ContainerConfigParseError() :
+        m_message("Container config parse error")
+    {
+    }
+
+    ContainerConfigParseError(const std::string &message) :
+        m_message(message)
+    {
+    }
+
+    virtual const char *what() const throw()
+    {
+        return m_message.c_str();
+    }
+
+protected:
+    std::string m_message;
+};
+
+class ContainerConfigParser
+{
+    LOG_DECLARE_CLASS_CONTEXT("CCPA", "ContainerConfigParser");
+
+public:
+    struct ContainerOptions {
+        bool enableWriteBuffer;
+    };
+
+    /**
+     * @brief Parse config needed for starting up the container in a correct manner.
+     *
+     * This reads a config string into json format, makes sure it is formatted
+     * correctly, and then calls readConfigElement for its elements.
+     *
+     * @param config a configuration string
+     */
+    ContainerOptions parseConfig(const std::string &config);
```

Comment:
I think "parse" would be enough for this name, the context etc. should always make the "Config" part redundant. A matter of taste ofc, but for what it's worth.